salon_creator/lib/screens/salon_creation_screen.dart

```diff
@@ -8,20 +20,529 @@ class SalonCreationScreen extends StatefulWidget {
 }
 
 class _SalonCreationScreenState extends State<SalonCreationScreen> {
+  bool isFilled = false;
+  bool inProgress = false;
+  final ImagePicker imagePicker = ImagePicker();
+  int index = 0;
+  List<XFile> _imageFiles = [];
+
+  ScrollController _scrollController = ScrollController();
+  TextEditingController categoriesController = TextEditingController();
+  TextEditingController descriptionController = TextEditingController();
+  TextEditingController priceController = TextEditingController();
+  TextEditingController salonNameController = TextEditingController();
+  XFile image;
+
+  void _updateScreenContext() {
+    var _hasProfileChanged = salonNameController.text.isNotEmpty &&
+        categoriesController.text.isNotEmpty &&
+        descriptionController.text.isNotEmpty &&
+        priceController.text.isNotEmpty &&
+        _imageFiles.isNotEmpty;
+    print(salonNameController.text.isNotEmpty);
+
+    print(priceController.text.isNotEmpty);
+    print(descriptionController.text.isNotEmpty);
+    print(categoriesController.text.isNotEmpty);
+    print(_imageFiles.isNotEmpty);
+    print(isFilled);
+
+    setState(() {
+      isFilled = _hasProfileChanged;
+    });
+  }
+
+  _scrollToRight() {
+    _scrollController.jumpTo(_scrollController.position.maxScrollExtent);
+  }
+
   @override
   Widget build(BuildContext context) {
+    final mq = MediaQuery.of(context).size;
+    final screenWidth = mq.width;
+    final screenHeight = mq.height;
+
     return Scaffold(
-      appBar: AppBar(
-        actions: [
-          TextButton(
-            onPressed: () {
-              Navigator.of(context).pushReplacementNamed('/home');
-            },
-            child: Text("スキップ"),
+      appBar: inProgress ? null : _buildAppBar(context),
```

Comment:
```suggestion
      appBar: operationInProgress ? null : _buildAppBar(context),
```
Rename the variable for better readability?